### ISSUE TITLE
Release 0.660 planning

### ISSUE THREAD

**Issue by msullivan** [python/mypy#6130]
I'd like to release mypy 0.660 on Friday, January 11. I plan to cut the release branch on Monday, January 7.

I plan to also release our first official `mypy_mypyc` binary alongside it.

If there are any PRs that need to make it in for the release, please mention them here.

**Comment by Michael0x2a** [comment]
I have a host of Literal-types related PRs I'd like to get in if possible.

- [Add intelligent indexing of tuples, NamedTuples, and TypedDict](https://github.com/python/mypy/pull/6124). This still needs code review.

- [Remove RawLiteralTypes synthetic type](https://github.com/python/mypy/pull/6121). I'm planning on either working on this PR or closing it and submitting a related one today.

- [Add interactions between Literal and Final](https://github.com/python/mypy/pull/6081). I'm planning on submitting an update for this PR later today.

- [Add tests for Literal types with incremental and fine-grained mode](https://github.com/python/mypy/pull/6075). This PR is blocked on the second bullet point; I'll update this once that one lands.

I'm also planning on submitting two more lower-priority PRs later tomorrow: one which improves error messages when you try constructing Literal types with things like arithmetic expressions or complex numbers, and one adding docs for literal types. I think it's ok to defer these two though -- the error message PR is mostly just a nice-to-have, and the docs don't need to make the monday cutoff.

**Comment by JukkaL** [comment]
If we ship a compiled binary by default, maybe the version should be incremented to 0.700. This is a major change, even if it doesn't bring any new functionality.

I might also want to make `--fast-exit` enabled by default.

**Comment by msullivan** [comment]
I hadn't been planning to make the compiled binary the default in this release yet, just to release and publicize an alternate compiled package.

**Comment by ethanhs** [comment]
I would like to get https://github.com/python/mypy/pull/6148 in if possible. This should bring the daemon on Windows to feature parity with Unix.

**Comment by ethanhs** [comment]
It would also be nice to get https://github.com/python/mypy/pull/6153 in since it fixes a rather confusing crash with the daemon.

**Comment by JukkaL** [comment]
> I hadn't been planning to make the compiled binary the default in this release yet, just to release and publicize an alternate compiled package.

Ok, 0.660 is good then and we don't need any `--fast-exit` changes.

**Comment by msullivan** [comment]
I am postponing the release to Wednesday, Jan 16. I'll cut the release branch later this week.

**Comment by Michael0x2a** [comment]
I have one final Literal-related PR I'd like to get in if possible. It should be a short one though: https://github.com/python/mypy/pull/6176

**Comment by msullivan** [comment]
I've cut the release branch

**Comment by msullivan** [comment]
We realized that it was a mistake to release typed_ast 1.1.1 with a breaking API change. As a result we are going to release a 1.2.0 and pull version 1.1.1, which means we need to get that out and update our dependency in mypy for this release.

**Comment by msullivan** [comment]
I'm cherry-picking #6191 because it is super low risk and might help reduce confusion of packagers?
